src/cartservice/src/cartstore/SpannerCartStore.cs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Google.Cloud.Spanner.Data;
+using Grpc.Core;
+using Microsoft.Extensions.Configuration;
+using System.Threading.Tasks;
+
+namespace cartservice.cartstore
+{
+    public class SpannerCartStore : ICartStore
+    {
+        private static readonly string TableName = "CartItems";
+        private readonly string databaseString;
+
+        public SpannerCartStore(IConfiguration configuration)
+        {
+            string spannerProjectId = configuration["SPANNER_PROJECT"];
+            SpannerConnectionStringBuilder builder = new();
+            builder.DataSource = $"projects/{spannerProjectId}/instances/hipstershop/databases/carts";
```

Comment:
The more I think about the Spanner instance name, here `hipstershop`, the more I think it should be an additional environment variable `SPANNER_NAME`, with that the end user could customize more easily the target of a specific instance. Think about DEV/TEST/PROD in same project, etc.

For the name of the database, we could let is hardcoded (or at least as a static string here) and document that the name of the database should be `carts` like we have it in the `.ddl` file.

Reply:
I agree that making instance configurable will be useful. Maybe for the scope of this PR, I can expose the value as an environmental variable, but have it default to `hipstershop` (or `onlineboutique`) if it's left undefined. That way, the quickstart remains simple and straightforward, but advanced users can go in and change the instance by editing the kubernetes manifest.

Sound ok?

Reply:
Yep, great! Let's do `SPANNER_NAME` as environment variable with the default value `onlineboutique`?

Reply:
Done: c3021bb20409f22b2c8f88e182ac44783c365d7b

Reply:
Like discussed offline, let's actually have just one env var `SPANNER_CONNECTION_STRING` which will get the value `projects/{spannerProjectId}/instances/{spannerInstanceId}/databases/carts`. More generic, and will propose more scenarios for the end users: DEV/TEST/QA/PROD where they could have multiple database under the same instance to minimize costs, etc.